tools/wasi/wasi.cpp

```diff
@@ -82,40 +85,56 @@ fizzy::ExecutionResult wasi_fd_prestat_get(fizzy::Instance& instance, const fizz
 
     uvwasi_serdes_write_prestat_t(instance.memory->data(), prestat_ptr, &buf);
 
-    return fizzy::Value{uint32_t{ret}};
+    return Value{uint32_t{ret}};
 }
 
-fizzy::ExecutionResult wasi_environ_sizes_get(
-    fizzy::Instance& instance, const fizzy::Value* args, int)
+ExecutionResult environ_sizes_get(Instance& instance, const Value* args, int)
 {
     const auto environc = args[0].as<uint32_t>();
     const auto environ_buf_size = args[1].as<uint32_t>();
     // TODO: implement properly (only returns 0 now)
     uvwasi_serdes_write_uint32_t(instance.memory->data(), environc, 0);
     uvwasi_serdes_write_uint32_t(instance.memory->data(), environ_buf_size, 0);
-    return fizzy::Value{uint32_t{UVWASI_ESUCCESS}};
+    return Value{uint32_t{UVWASI_ESUCCESS}};
 }
+}  // namespace
 
-bool run(int argc, const char** argv)
+std::optional<bytes> load_wasm_binary(const std::filesystem::path& path, std::ostream& err)
```

Comment:
This function has pretty bad API, but I'm not sure how exactly improve it right now.
Also current error reporting is in some middle ground: it provides some additional but not complete set of diagnostics. And finally, the `std::ifstream` can sometimes be in "fail" state but it can also throw an exception in other cases...

Reply:
Not sure how `filesystem::path` works but maybe this function should take a string (provided by user) and create `filesytem::path` inside, handling the errors like invalid symbols in path.

Reply:
I considered the following options and used the one which is the easiest to use at the moment.
1. `const char*` - only literals and `argv[]` work really,
2. `const std::string&` - specific string type used, conversion from `const char*` is needed.
3. `std::string_view` - does not work with `std::ifstream` - conversion to `std::string` is needed.
4. `std::filesystem::path` - works with everything, emphasize that this is path.

I agree that invalid path format should be also handled inside. In the some time, there are other error cases that are not handled explicitly.

So switch to some other type?

Reply:
(I imagined it like taking `string_view` then constucting `path` out of it, and then `ifstream` out of this `path`)

I'm fine to leave it like this for now. At some point exceptions thrown by `path` constructor will need to be handled (when we address `@todo Make noexcept.`)

Reply:
Done.